### PR TITLE
{2023.06}[foss/2021a] Qt5 V5.15.2

### PR DESCRIPTION
**Pull request by TopRichard**

### CODE REVIEW

**Comment by TopRichard** [comment]
bot: build repo:eessi-2023.06-software arch:x86_64/generic

**Comment by TopRichard** [comment]
bot: build repo:eessi-2023.06-software arch:x86_64/generic

**Comment by TopRichard** [comment]
bot: build repo:eessi-2023.06-software arch:x86_64/generic

**Comment by TopRichard** [comment]
Failure : build failed to determine installation prefix for binutils

**Comment by boegel** [comment]
Sanity check failed:
```
== sanity checking...
== ... (took < 1 sec)
3:35:06  28 out of 29 easyconfigs done: libGLU/9.0.1-GCCcore-10.3.0 (OK), Mesa/2ERROR: Traceback (most recent call last):
  File "/cvmfs/pilot.eessi-hpc.org/versions/2023.06/software/linux/x86_64/generic/software/EasyBuild/4.7.2/lib/python3.11/site-packages/easybuild/main.py", line 133, in build_and_install_software
    (ec_res['success'], app_log, err) = build_and_install_one(ec, init_env)
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/cvmfs/pilot.eessi-hpc.org/versions/2023.06/software/linux/x86_64/generic/software/EasyBuild/4.7.2/lib/python3.11/site-packages/easybuild/framework/easyblock.py", line 4236, in build_and_install_one
    result = app.run_all_steps(run_test_cases=run_test_cases)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/cvmfs/pilot.eessi-hpc.org/versions/2023.06/software/linux/x86_64/generic/software/EasyBuild/4.7.2/lib/python3.11/site-packages/easybuild/framework/easyblock.py", line 4119, in run_all_steps
    self.run_step(step_name, step_methods)
  File "/cvmfs/pilot.eessi-hpc.org/versions/2023.06/software/linux/x86_64/generic/software/EasyBuild/4.7.2/lib/python3.11/site-packages/easybuild/framework/easyblock.py", line 3954, in run_step
    step_method(self)()
  File "/cvmfs/pilot.eessi-hpc.org/versions/2023.06/software/linux/x86_64/generic/software/EasyBuild/4.7.2/lib/python3.11/site-packages/easybuild/easyblocks/q/qt.py", line 203, in sanity_check_step
    if LooseVersion(glibc_version) <= LooseVersion("2.16"):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/cvmfs/pilot.eessi-hpc.org/versions/2023.06/compat/linux/x86_64/usr/lib/python3.11/site-packages/setuptools/_distutils/version.py", line 78, in __le__
    c = self._cmp(other)
        ^^^^^^^^^^^^^^^^
  File "/cvmfs/pilot.eessi-hpc.org/versions/2023.06/compat/linux/x86_64/usr/lib/python3.11/site-packages/setuptools/_distutils/version.py", line 351, in _cmp
    if self.version < other.version:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: '<' not supported between instances of 'str' and 'int'
```

So it seems like `glibc_version` is not a valid version for some reason...

**Comment by boegel** [comment]
Indeed:
```
$ source /cvmfs/pilot.eessi-hpc.org/versions/2023.06/init/bash
$ module load EasyBuild/4.7.2
$ eb --show-system-info | grep glibc
  -> glibc version: UNKNOWN
```

Or more directly:

```
$ python -c 'from easybuild.tools.systemtools import get_glibc_version; print(get_glibc_version())'
UNKNOWN
```
```

**Comment by boegel** [comment]
As a workaround, we can set `check_qtwebengine` to `False` through a hook before the `sanity_check_step` runs.

It doesn't affect the installation itself, only the sanity check part, where the `glibc` version is being checked to verify whether `libQtWebEngine` should be there, or not (it should be, since we have a very recent glibc).

**Comment by boegel** [comment]
`glibc` version in EESSI compat layer will be detected correctly by EasyBuild once https://github.com/easybuilders/easybuild-framework/pull/4290 is merged

**Comment by TopRichard** [comment]
@boegel Can we use the option:ignore-test-failure as it is the only test within the easyconfig file ?

**Comment by TopRichard** [comment]
bot: build repo:eessi-2023.06-software arch:x86_64/generic

**Comment by boegel** [comment]
> @boegel Can we use the option:ignore-test-failure as it is the only test within the easyconfig file ?

The `--ignore-test-failure` option is to ignore failing tests during the test step.
Here the problem is that the sanity check step is failing.

If you need help with implementing a hook to dance around this, let me know...

**Comment by TopRichard** [comment]
bot: build repo:eessi-2023.06-software arch:x86_64/generic

**Comment by TopRichard** [comment]
bot: build repo:eessi-2023.06-software arch:x86_64/generic

**Comment by TopRichard** [comment]
bot: build repo:eessi-2023.06-software arch:x86_64/intel/haswell
bot: build repo:eessi-2023.06-software arch:x86_64/intel/skylake_avx512
bot: build repo:eessi-2023.06-software arch:x86_64/amd/zen3
bot: build repo:eessi-2023.06-software arch:x86_64/amd/zen2
bot: build repo:eessi-2023.06-software arch:aarch64/generic
bot: build repo:eessi-2023.06-software arch:aarch64/neoverse_v1
bot: build repo:eessi-2023.06-software arch:aarch64/neoverse_n1

File: eb_hooks.py [hunk 1]
```diff
@@ -59,9 +59,16 @@ def parse_hook(ec, *args, **kwargs):
     # determine path to Prefix installation in compat layer via $EPREFIX
     eprefix = get_eessi_envvar('EPREFIX')
 
+#Disable Qt5/5.15.2 qtwebengine in Sanity-check
+    Qt5_check_qtwebengine_disable(ec, eprefix)
+
     if ec.name in PARSE_HOOKS:
         PARSE_HOOKS[ec.name](ec, eprefix)
 
+def Qt5_check_qtwebengine_disable(ec, eprefix):
```
Comment (boegel) [inline eb_hooks.py:68]: @TopRichard Let's rename this to `parse_hook_Qt5_check_qtwebengine_disable`, and register it in `PARSE_HOOKS`, so we keep things consistent.

Done in https://github.com/TopRichard/bot-software-layer1/pull/13

**Review by boegel — CHANGES_REQUESTED** [review]

File: eessi-2023.06-eb-4.7.2-2021a.yml [hunk 2]
```diff
@@ -8,3 +8,6 @@ easyconfigs:
   - libjpeg-turbo-2.0.6-GCCcore-10.3.0.eb
   - QuantumESPRESSO-6.7-foss-2021a.eb
   - GROMACS-2021.3-foss-2021a.eb
+  - Qt5-5.15.2-GCCcore-10.3.0.eb:
+       options:
```
Comment (boegel) [inline eessi-2023.06-eb-4.7.2-2021a.yml:12]: Let's add a comment to explain this:

```suggestion
       # add missing patch files for Qt5 5.15.2 to fix build problems with glibc 2.34,
       # see https://github.com/easybuilders/easybuild-easyconfigs/pull/18087
       options:    
```

**Review by boegel — CHANGES_REQUESTED** [review]

File: eessi-2023.06-eb-4.7.2-2021a.yml [hunk 3]
```diff
@@ -8,3 +8,8 @@ easyconfigs:
   - libjpeg-turbo-2.0.6-GCCcore-10.3.0.eb
   - QuantumESPRESSO-6.7-foss-2021a.eb
   - GROMACS-2021.3-foss-2021a.eb
+  - Qt5-5.15.2-GCCcore-10.3.0.eb:
+# add missing patch files for Qt5 5.15.2 to fix build problems with glibc 2.34,
+# see https://github.com/easybuilders/easybuild-easyconfigs/pull/18087
```
Comment (boegel) [inline eessi-2023.06-eb-4.7.2-2021a.yml:13]: @TopRichard Can we match the indent with `options:`, or does that cause trouble?

**Review by boegel — DISMISSED** [review]
lgtm

**Comment by boegel** [comment]
@TopRichard Let's trigger a rebuild for (only) `x86_64/generic` to make sure the refactored hooks are working, before we trigger a deploy?

**Comment by TopRichard** [comment]
bot: build repo:eessi-2023.06-software arch:x86_64/generic

**Comment by boegel** [comment]
All tarballs staged for ingestion

**Review by boegel — APPROVED** [review]

**Pull request merged**